feature-libs/cart/main/core/facade/selective-cart.service.ts

```diff
@@ -1,199 +1,134 @@
 import { Injectable } from '@angular/core';
-import { Store } from '@ngrx/store';
 import {
   Cart,
+  CartType,
   OrderEntry,
   SelectiveCartFacade,
 } from '@spartacus/cart/main/root';
 import {
   BaseSiteService,
   OCC_USER_ID_ANONYMOUS,
-  StateUtils,
   UserIdService,
   UserService,
 } from '@spartacus/core';
-import { BehaviorSubject, combineLatest, Observable, of } from 'rxjs';
-import { filter, map, shareReplay, switchMap, take, tap } from 'rxjs/operators';
-import { StateWithMultiCart } from '../store/multi-cart-state';
+import { combineLatest, Observable } from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  shareReplay,
+  switchMap,
+  take,
+  tap,
+  withLatestFrom,
+} from 'rxjs/operators';
 import { MultiCartService } from './multi-cart.service';
 
 @Injectable()
 export class SelectiveCartService implements SelectiveCartFacade {
-  protected customerId: string;
-  protected userId: string;
-  protected cartId: string;
   protected selectiveCart$: Observable<Cart>;
-  protected cartId$: BehaviorSubject<string> = new BehaviorSubject<string>('');
-
-  protected readonly PREVIOUS_USER_ID_INITIAL_VALUE =
-    'PREVIOUS_USER_ID_INITIAL_VALUE';
-  protected previousUserId = this.PREVIOUS_USER_ID_INITIAL_VALUE;
-
-  protected cartSelector$ = this.cartId$.pipe(
-    switchMap((cartId) => {
-      this.cartId = cartId;
-      return this.multiCartService.getCartEntity(cartId);
-    })
-  );
 
   constructor(
-    protected store: Store<StateWithMultiCart>,
     protected userService: UserService,
     protected multiCartService: MultiCartService,
     protected baseSiteService: BaseSiteService,
     protected userIdService: UserIdService
-  ) {
-    combineLatest([
-      this.userService.get(),
-      this.baseSiteService.getActive(),
-    ]).subscribe(([user, activeBaseSite]) => {
-      if (user && user.customerId && activeBaseSite) {
-        this.customerId = user.customerId;
-        this.cartId$.next(`selectivecart${activeBaseSite}${this.customerId}`);
-      } else if (user && !user.customerId) {
-        this.cartId$.next('');
-      }
-    });
-
-    this.userIdService.getUserId().subscribe((userId) => {
-      this.userId = userId;
-
-      if (this.isJustLoggedIn(userId)) {
-        this.load();
-      }
-
-      this.previousUserId = userId;
-    });
-
-    this.selectiveCart$ = this.cartSelector$.pipe(
-      map(
-        (
-          cartEntity: StateUtils.LoaderState<Cart>
-        ): {
-          cart: Cart;
-          loading: boolean;
-          loaded: boolean;
-        } => {
-          return {
-            cart: cartEntity.value as Cart,
-            loading: Boolean(cartEntity.loading),
-            loaded: Boolean(
-              (cartEntity.error || cartEntity.success) && !cartEntity.loading
-            ),
-          };
-        }
-      ),
-      filter(({ loading }) => !loading),
```

Comment:
Do we still need to filter out loading carts?

Reply:
in our new code, we don't care the cart is loading or not. If there is `selectiveId` in index, we just use this `id` to read the cart data.